hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFsShell.java

```diff
@@ -0,0 +1,590 @@
+        shellRun("-mkdir", "-p", testDir + "/subdir1/subdir2"));
+    assertTrue("Directory should exist",
+        fs.getFileStatus(new Path(testDir, "subdir1/subdir2")).isDirectory());
+
+    // create a new bucket with hadoop fs will return error file exists
+    // because innerGetFileStatus return root directory status without a probe
+    String newBucketName =
+        "hadoop-fs-shell-test-" + ThreadLocalRandom.current().nextLong(Long.MAX_VALUE);
+    assertNotEquals("Should not be able to create new bucket", 0,
+        shellRun("-mkdir", "s3a://" + newBucketName + "/"));
+    assertTrue(err.toString().contains("mkdir: `s3a://" + newBucketName + "/': File exists"));
+    err.reset();
+
+    assertEquals("Should list directory success", 0, shellRun("-ls", testDir.toString()));
+    assertTrue("Should found one item", out.toString().contains("Found 1 items"));
+    assertTrue("Should print file list to stdout", out.toString().contains(testDir + "/subdir1"));
+    out.reset();
+
+    assertEquals("Should list directory as a plain file success", 0,
+        shellRun("-ls", "-d", testDir.toString()));
+    assertTrue("Should print directory path to stdout",
+        out.toString().contains(testDir.toString()));
+    out.reset();
+
+    assertNotEquals("Should fail when list with display erasure coding policy flag", 0,
+        shellRun("-ls", "-e", testDir.toString()));
+    assertTrue(err.toString()
+        .contains("FileSystem " + fs.getUri().toString() + " does not support Erasure Coding"));
+    err.reset();
+
+    assertEquals("Should recursively list directory success", 0,
+        shellRun("-ls", "-R", testDir.toString()));
+    assertTrue("Should print file list to stdout", out.toString().contains(testDir.toString()));
+    assertTrue("Should print file list to stdout", out.toString().contains(testDir + "/subdir1"));
+    assertTrue("Should print file list to stdout",
+        out.toString().contains(testDir + "/subdir1/subdir2"));
+    out.reset();
+
+    assertEquals("Should delete directory success", 0,
+        shellRun("-rmdir", testDir + "/subdir1/subdir2"));
+    assertFalse("Directory should not exist", fs.exists(new Path(testDir, "subdir1/subdir2")));
+
+    assertNotEquals("Should not be able to delete non-empty directory", 0,
+        shellRun("-rmdir", testDir.toString()));
+    assertTrue(err.toString().contains("Directory is not empty"));
+    assertTrue("Directory should exist", fs.exists(testDir));
+    err.reset();
+
+    assertEquals("Should recursively delete directory success", 0,
+        shellRun("-rm", "-r", testDir.toString()));
+    assertFalse("Directory should not exist", fs.exists(testDir));
+
+    assertNotEquals("Should not be able to delete root directory", 0,
+        shellRun("-rm", "-r", "-f", fs.getUri().toString() + "/"));
+    assertTrue(err.toString().contains("Input/output error"));
+    err.reset();
+  }
+
+  @Test
+  public void testFsShellFileOperations() throws IOException {
```

Comment:
This method is too long, and raises a checkstyle violation.

```
./hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFsShell.java:141:  @Test:3: Method length is 155 lines (max allowed is 150). [MethodLength]
```

Can we break it down? Maybe move the stream concatenation (cat, head, tail) to another method?

Reply:
Good suggestion, I will break them to another method.